### PR TITLE
refactor!: modularization

### PR DESCRIPTION
**Pull request by saul-jb**
This PR is a work in progress example of refactoring the registrar into a more modular design.

I envision the end API to look something like this, obviously taking inspiration for how Libp2p/Helia are structured:
```javascript
import { createWelo } from 'welo'
import { liveReplicator } from 'welo/replicator/live'
import { basalEntry, createBasalEntryConfig } from 'welo/entry/basal'
import { basalIdentity, createBasalIdentityConfig } from 'welo/identity/basal'
import { staticAccess, createStaticAccessConfig } from 'welo/access/static'
import { keyvalue, createKeyvalueConfig } from 'welo/store/keyvalue'

const welo = await createWelo({
  helia,
  datastore: LevelDatastore,
  replicators: [ liveReplicator() ],
  identities: [ basalIdentity() ],
  entries: [ basalEntry() ],
  access: [ staticAccess() ]
})

const manifest = await welo.determine({
  name: "my-database",
  access: createStaticAccessConfig({ write: ['*'] }),
  entry: createBasalEntryConfig(),
  store: createKeyvalueConfig(),
  identity: createBasalIdentityConfig()
})

const db = await welo.open(manifest)
const store = await keyvalue(db)
```

I would like to simplify (shorten) the code needed to create a manifest but I'm not sure how yet.

### CODE REVIEW

File: src/entry/interface.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import type { BlockView } from 'multiformats/interface'
 import type { CID } from 'multiformats/cid'
 
 import type { Blocks } from '@/blocks/index.js'
+import { OPALSNT_PREFIX } from '@/utils/constants.js'
```
Comment (tabcat) [inline src/entry/interface.ts:5]: This variable needs to be changed to `HLDB_PREFIX` in constants.js. I thought I had done this already. Feel free to do it as part of this PR or I can do it after merging.

**Review by tabcat — COMMENTED** [review]
Loving the changes so far :+1: 

:crab: registry gone :crab: 

These will require a major version bump so add a ! to the end of refactor or I can do that when merging

File: src/utils/index.ts [hunk 2]
```diff
@@ -59,12 +60,9 @@ interface Components {
...
-): Components => ({
-  Access: registry.access.get(manifest.access.protocol),
-  Entry: registry.entry.get(manifest.entry.protocol),
-  Identity: registry.identity.get(manifest.identity.protocol),
-  Store: registry.store.get(manifest.store.protocol)
+export const getComponents = (): Components => ({
+  Access: StaticAccess,
+  Store: Keyvalue,
+  Entry,
+  Identity
```
[the remark this answers is not in the document]
Comment (tabcat) [reply src/utils/index.ts:67]: Can add this later

File: src/entry/interface.ts (hunk 1 above)
Comment (saul-jb) [reply src/entry/interface.ts]: Alright, I've added the change to this fork.

File: src/utils/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (saul-jb) [reply src/utils/index.ts]: This has now been fixed, there was (and still is) some things I have just modified solely for the tests to pass while I am changing other things.

File: src/utils/index.ts [hunk 3]
```diff
@@ -33,22 +35,21 @@ export const dirs = (root: string): DirsReturn =>
...
+    protocol: StaticAccess.protocol,
     config: { write: [identity.id] }
   },
   entry: {
-    protocol: registry.entry.star.protocol
+    protocol: Entry.protocol
   },
   identity: {
-    protocol: registry.identity.star.protocol
+    protocol: Identity.protocol
```
[the remark this answers is not in the document]
Comment (saul-jb) [reply src/utils/index.ts]: I have added this change, which should also help make the manifest creation code shorter.

**Comment by saul-jb** [comment]
Register is now completely gone along with the multi-replicator. More changes to come.

**Comment by saul-jb** [comment]
Unfortunately this has become a 'rewrite the entire codebase' PR, so apologies for that, it should have probably been split into a few PRs.

Major changes:
- Register has been removed.
- Multi-replication is the default and multi-replicator removed.
- Modules have been refactored for a more modular design.
- Level-datastores have been replaced with namespaced datastore instances.
- Static modules have been replaced module instances.
- Types have been cleaned up a bit.

Marking this as ready but will need reviewing to ensure that functionality remains the same.

**Review by tabcat — CHANGES_REQUESTED** [review]
Looking really good :fire: :+1: :100: just a few (relative) things to talk about

Love the use of `NamespaceDatastore` !!

wondering if we could use the helia data/block -stores (by default) if they are name spaced

Feel free to choose address some of the comments in a new PR, fine merging everything as is

A change I'm thinking about has to do with making it so `identityModule` can be just `identity` inside `database.modules` or `database.components` and `database.identity` can become `database.writer` to be sure there is no confusion (this is for a future PR one of us makes)

Also would like to (in the near future) make creating a welo instance simpler and show that in the usage, helia just did a default helia config for createHelia that does this (for reference)

**Comment by saul-jb** [comment]
> wondering if we could use the helia data/block -stores (by default) if they are name spaced

I suspect this is a bad idea since Helia would be expecting exclusive access to the store, instead I used the in-memory store as the default (like Helia does) and if the user wants to use the same store for both they can pass unique namespaces to both Welo and Helia.

**Pull request merged**